### PR TITLE
finished rewrite of barracks radius

### LINKED ISSUE #280: Can not get Sound and Crash
### Minecolonies version

minecolonies-universal-1.10.2-0.4.1719.jar

### Expected behavior

- The builder can create a building :)

### Actual behaviour

- Crashing while playing a sound :(

### Steps to reproduce the problem

1. Create a Colony
2. Start a new Buildung
3. The Builder want to remove something
4. Crash, see Log
[crash-2016-11-13_09.15.34-server.txt](https://github.com/Minecolonies/minecolonies/files/587646/crash-2016-11-13_09.15.34-server.txt)

The easist way to fix is, at a try catch around the sound at com.minecolonies.entity.EntityCitizen.hitBlockWithToolInHand(EntityCitizen.java:1386) ;)

### LINKED ISSUE #280: Barracks gives slower than expected progression to increase in colony border size
### Prerequisites 

- [X ] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [ X] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [X ] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.16.4
- MineColonies Version: 0.13.596-ALPHA-universal
- Structurize Version: 0.13.113-ALPHA-universal
- Related Mods and their Versions:

## Expected behavior

Barracks, having multiple towers and many guards, should increase the colony border at least as well, level wise, as guard towers.

## Actual behavior

Barracks divide your tower levels by 4 to calculate radius.  meaning that after the 1 chunk given from initial build of barracks, you must level the barracks to 3 and towers to 3 (except one can be 2) for a total of 8 levels to divide by 4 and get your second chunk radius bonus.

border

This screenshot is the proximity of the border to a barracks.  (radius 1)

## Steps to reproduce the issue

1. build barracks in the corner of your colony.
2. upgrade towers until colony border increases

### Logs

- latest.log: 
- crashlog: 

### Notes

Perhaps this should be a "feature" request, but it seems like perhaps it was legacy code that didn't account for lower level barracks having fewer towers.
Barracks are so much more difficult to level up than guard towers, and have many guards.  I believe that the progression makes much more sense to have a maxxed out barracks have radius equal to its level.  this makes it comparable to a guard tower.

Suggested:
radius 1 = level 1 barracks with 1,NA,NA,NA
radius 2 = level 2 barracks with 2,2,NA,NA
radius 3 = level 3 barracks with 3,3,3,NA
radius 4 = level 4 barracks with 4,4,4,4
radius 5 = level 5 barracks with 5,5,5,5

Currently:
radius 1 = level 1 barracks with 0,NA,NA,NA
radius 2 = level 3 barracks with 3,3,2,NA
radius 3 = level 4 barracks with 3,3,3,3
radius 4 = level 4 barracks with 4,4,4,4
radius 5 = level 5 barracks with 5,5,5,5

This is a strange non-linear progression, that isn't well aligned with barracks level.  You'll notice that the top tier radii are not changed, just the entry level ones.

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by Vaevictusnet**
Closes https://github.com/ldtteam/minecolonies-features/issues/280

# Changes proposed in this pull request:
As per discussion with Raycoms and SomeAddons, this simplifies the colony border of the Barracks to 2 unless all 4 towers and the barracks are level 4+, which increases to 3.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBarracks.java [hunk 1]
```diff
@@ -196,16 +196,21 @@ public int getClaimRadius(final int newLevel)
             return 0;
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBarracks.java:196]: return 1 here, no?

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBarracks.java [hunk 2]
```diff
@@ -196,16 +196,21 @@ public int getClaimRadius(final int newLevel)
...
+        int barracksClaimRadius = 3;
         for (final BlockPos pos : towers)
         {
             final IBuilding building = colony.getBuildingManager().getBuilding(pos);
             if (building != null)
             {
-                sum += building.getBuildingLevel();
+                if (building.getBuildingLevel() < 4) 
+                { 
+                  barracksClaimRadius = 2;
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBarracks.java]: the indent here seems of a little bit

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBarracks.java (hunk 1 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBarracks.java:196]: no level 0 buildings shouldnt claim anything around them^^

**Review by someaddons — APPROVED** [review]

**Comment by ravenbuilder934** [comment]
Okay, now someone needs to make a wiki issue/PR with the fixed table, and I'll merge it

**Pull request merged**